### ISSUE TITLE
PSPNet blank prediction result

### ISSUE THREAD

**Issue by mesakh123** [qubvel/segmentation_models.pytorch#385]
Sorry for my broke english

I was trying to train my model using Unet and PSPnet, the both models result of iou and loss scores are pretty, but when i tried to do prediction with my trained model, i found out that pspnet model prediction result is blank/empty, i was make sure to make the all parameters are correct. Since i was train the model of multi classes, both softmax2d and softmax  activations were tried, the result still blank/empty. As mentioned on #176  , looks like activation by torch implementation as the main reason, When i used only one class with sigmoid activation, the PSPnet and Unet model works well. But when using softmax as activation, PSPnet can't predict correctly. If you have any suggestion, please comment, thank you. 

my ground truth and prediction result (PSPNet) : 
![image](https://user-images.githubusercontent.com/28601218/115431207-a0d2c280-a237-11eb-9068-bb24d2a8f62c.png)

the multiclass implementation similar with the code below :
[https://github.com/shirokawakita/multiclass-segmentation/blob/main/example_camvid_multiclassB_quita.ipynb](https://github.com/shirokawakita/multiclass-segmentation/blob/main/example_camvid_multiclassB_quita.ipynb)

**Comment by mesakh123** [comment]
After a week's research and trying many different repos of PSPNet with pytorch, I found out the way to solve the problem. PSPNet maybe not good enough when the class with ground truth mask size is too small. Just ignore the background classes to calculate metrics and losses. Training with ignoring the background could lead to higher loss, even overfitting, depends on the dataset. On the other hand, UNet and Deeplabv3 work fine even with background channel classes. Also PSPNet at least need some activation such as sigmoid or softmax2d, so that the loss will not become negative. If you guys have found the way to train PSPNet without ignoring background class, please tell me, thank you.
